GoogleForCreators/web-stories-wp
# Editor: Virtualize Text Sets List

## Summary

Adds virtualization to the Text Sets list to improve performance.

## Relevant Technical Choices

- Chose `react-virtual` as a headless virtualization solution. 
- B/c `react-virtual` requires a ref to the outer most container of the list, I used a `ResizeObserver` in the parent of TextSets (TextPane) to force updates to it's child (TextSets) when the height changes.  This ensures that the ref that is passed as a prop is always up to date.

## User-facing changes

- None

## Testing Instructions

- Launch into the editor and verify that the text sets pane works.  It feels especially snappy in production mode.
---



Partial #4799 as virtualization will likely be used in the dashboard
